core/services/bulletprooftxmanager/eth_broadcaster.go

```diff
@@ -58,80 +56,67 @@ type ethBroadcaster struct {
 	ethClient eth.Client
 	config    orm.ConfigReader
 
-	started    bool
-	stateMutex sync.RWMutex
-
-	ethTxInsertListener *utils.PostgresEventListener
+	ethTxInsertListener postgres.Subscription
+	eventBroadcaster    postgres.EventBroadcaster
 
 	// trigger allows other goroutines to force ethBroadcaster to rescan the
 	// database early (before the next poll interval)
 	trigger chan struct{}
 	chStop  chan struct{}
 	wg      sync.WaitGroup
+
+	utils.StartStopOnce
 }
 
 // NewEthBroadcaster returns a new concrete ethBroadcaster
-func NewEthBroadcaster(store *store.Store, config orm.ConfigReader) EthBroadcaster {
+func NewEthBroadcaster(store *store.Store, config orm.ConfigReader, eventBroadcaster postgres.EventBroadcaster) EthBroadcaster {
```

Comment:
I know we're trying to cut down the scope of what we inject. Would it be worth just passing in the subscription instead of passing around the whole event broadcaster?

Reply:
We could, but I encountered two problems with that:
1) subscribing in application.go seems strange
2) `pipeline.ORM` has to create and destroy subscriptions continually so it needs the EventBroadcaster regardless

I'm curious about
> trying to cut down the scope of what we inject

I've been actively trying to increase the scope what we inject (the above is a pattern that's worked well for me in the past, particularly when it comes to testing).  Should we circle up on this?